crates/data_components/src/github.rs

```diff
@@ -63,19 +65,35 @@ impl GithubFilesTableProvider {
         repo: &str,
         tree_sha: &str,
         include: Option<Arc<GlobSet>>,
+        fetch_content: bool,
     ) -> Result<Self> {
-        let schema = Arc::new(Schema::new(vec![
+        let mut fields = vec![
             Field::new("name", DataType::Utf8, true),
             Field::new("path", DataType::Utf8, true),
             Field::new("size", DataType::UInt64, true),
             Field::new("sha", DataType::Utf8, true),
             Field::new("mode", DataType::Utf8, true),
             Field::new("url", DataType::Utf8, true),
-        ]));
+            Field::new("download_url", DataType::Utf8, true),
+        ];
+
+        if fetch_content {
+            fields.push(Field::new("content", DataType::Utf8, true));
```

Comment:
I don't think it makes sense to change the schema based on whether its accelerated or not.